app/scripts/views/sign_in_recovery_code.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AuthErrors = require('../lib/auth-errors');
+const Cocktail = require('cocktail');
+const FormView = require('./form');
+const SignInMixin = require('./mixins/signin-mixin');
+const ServiceMixin = require('./mixins/service-mixin');
+const Template = require('templates/sign_in_recovery_code.mustache');
+const VerificationReasonMixin = require('./mixins/verification-reason-mixin');
+
+const CODE_INPUT_SELECTOR = 'input.recovery-code';
+const VALID_RECOVERY_CODE_REGEX = /^([a-zA-Z0-9]{10})|([a-fA-F0-9]{8})/;
+
+const View = FormView.extend({
+  className: 'sign-in-recovery-code',
+  template: Template,
+
+  beforeRender() {
+    const account = this.getSignedInAccount();
+    if (! account || ! account.get('sessionToken')) {
+      this.navigate(this._getAuthPage());
+    }
+  },
+
+  submit() {
+    const account = this.getSignedInAccount();
+    const code = this.getElementValue('input.recovery-code').toLowerCase();
+    return Promise.resolve()
+      .then(() => {
+        if (! VALID_RECOVERY_CODE_REGEX.test(code)) {
```

Comment:
One suggestion rather than doing inline validation here is to set up a new element type, see [email-input](https://github.com/mozilla/fxa-content-server/blob/c3a2d605202e1d2f8ec85e77a160dc64aeb200e0/app/scripts/views/elements/email-input.js#L12) for an example. There are a few of advantages to doing so:

* Once an element type is defined, its validation rules will be invoked automatically in every view as long as the `match` function matches a selector on one of that view's elements.
* Validation logic is removed from the view.
* You can define a `val` function that trims values transparently w/o the view needing to worry about doing so.

This approach would also be useful for places like the TOTP code where it can be entered in multiple places and you'll want the same validation/cleanup logic used in both.